Source/ARTMessage.m

```diff
@@ -45,3 +49,28 @@ - (NSInteger)messageSize {
 }
 
 @end
+
+@implementation ARTMessage (Decryption)
+
++ (instancetype)fromEncodedJsonObject:(NSDictionary *)input channelOptions:(ARTChannelOptions *)options error:(NSError **)error {
+    ARTJsonLikeEncoder *jsonEncoder = [[ARTJsonLikeEncoder alloc] initWithDelegate:[[ARTJsonEncoder alloc] init]];
+    ARTDataEncoder *decoder = [[ARTDataEncoder alloc] initWithCipherParams:options.cipher error:error];
+    
+    ARTMessage *message = [jsonEncoder messageFromDictionary:input];
+    message = [message decodeWithEncoder:decoder error:error];
+    
+    return message;
+}
+
++ (NSArray<ARTMessage *> *)fromEncodedJsonArray:(NSArray<NSDictionary *> *)jsonArray channelOptions:(ARTChannelOptions *)options error:(NSError **)error {
+    ARTJsonLikeEncoder *jsonEncoder = [[ARTJsonLikeEncoder alloc] initWithDelegate:[[ARTJsonEncoder alloc] init]];
+    ARTDataEncoder *decoder = [[ARTDataEncoder alloc] initWithCipherParams:options.cipher error:error];
+    
+    NSArray <ARTMessage *> *messagesArray = [jsonEncoder messagesFromArray:jsonArray];
+    messagesArray = [messagesArray artMap:^(ARTMessage *message) {
+        return [message decodeWithEncoder:decoder error:nil];
```

Comment:
Are you sure we can't manage this error? It's kind of risky to leave it as nil especially when there is a requirement where we should pass a pointer to it.

Reply:
You're right, I should do something about it) `artMap:` closure didn't allowed me to refer `error`, but it's a no excuse.

Reply:
https://github.com/ably/ably-cocoa/pull/1307/commits/caf46b333fd9e559641e19b1a0e15ddf79248696